tests/unit_tests/grid/test_points_functions.py

```diff
@@ -0,0 +1,413 @@
+                                         [1600., 2000., 3020.], [1700., 2000., 3020.], [1800., 2000., 3020.]],
+                                        [[1000., 2100., 3020.], [1100., 2100., 3021.], [1200., 2100., 3021.],
+                                         [1300., 2100., 3020.], [1400., 2100., 3016.5], [1500., 2100., 3021.],
+                                         [1600., 2100., 3021.], [1700., 2100., 3021.], [1800., 2100., 3020.]],
+                                        [[1000., 2200., 3020.], [1100., 2200., 3023.5], [1200., 2200., 3023.5],
+                                         [1300., 2200., 3023.5], [1400., 2200., 3020.], [1500., 2200., 3023.5],
+                                         [1600., 2200., 3023.5], [1700., 2200., 3023.5], [1800., 2200., 3020.]],
+                                        [[1000., 2300., 3020.], [1100., 2300., 3019.], [1200., 2300., 3019.],
+                                         [1300., 2300., 3019.], [1400., 2300., 3016.5], [1500., 2300., 3020.],
+                                         [1600., 2300., 3019.], [1700., 2300., 3019.], [1800., 2300., 3020.]],
+                                        [[1000., 2400., 3020.], [1100., 2400., 3020.], [1200., 2400., 3020.],
+                                         [1300., 2400., 3019.], [1400., 2400., 3016.5], [1500., 2400., 3021.],
+                                         [1600., 2400., 3020.], [1700., 2400., 3020.], [1800., 2400., 3020.]],
+                                        [[1000., 2500., 3020.], [1100., 2500., 3020.], [1200., 2500., 3020.],
+                                         [1300., 2500., 3020.], [1400., 2500., 3020.], [1500., 2500., 3020.],
+                                         [1600., 2500., 3020.], [1700., 2500., 3020.], [1800., 2500., 3020.]]])
+
+    # Act
+    horizon_points = pf.horizon_points(faulted_grid, ref_k0 = 1)
+
+    # Assert
+    np.testing.assert_array_almost_equal(horizon_points, expected_horizon_points)
+
+
+def test_x_section_corner_points_default(basic_regular_grid):
+    # Arrange
+    expected_x_section_corner_points = np.array([[[[[0.0, 0.0, 0.0], [0.0, 50.0, 0.0]],
+                                                   [[0.0, 0.0, 20.0], [0.0, 50.0, 20.0]]],
+                                                  [[[0.0, 50.0, 0.0], [0.0, 100.0, 0.0]],
+                                                   [[0.0, 50.0, 20.0], [0.0, 100.0, 20.0]]]],
+                                                 [[[[0.0, 0.0, 20.0], [0.0, 50.0, 20.0]],
+                                                   [[0.0, 0.0, 40.0], [0.0, 50.0, 40.0]]],
+                                                  [[[0.0, 50.0, 20.0], [0.0, 100.0, 20.0]],
+                                                   [[0.0, 50.0, 40.0], [0.0, 100.0, 40.0]]]]])
+
+    # Act
+    x_section_corner_points = pf.x_section_corner_points(basic_regular_grid, axis = 'I')
```

Comment:
Please add a similar test for axis = 'J' (in that case the x values should vary, with y values constant).